Feature Updates

AzukaChan opened the pull request:
Refresh Torrent Data
----------------
Added a button to manually refresh torrent seed/leech data. There is currently no limit on this so you probably shouldn't go spamming it.


Category Overview
-----------------
Added a 'Category Overview' which can be accessed by clicking a category from the index page. Once a category reaches 10 torrents; previous/next buttons will appear at the bottom of the page allowing users to navigate through pages.


Basic Account/Admin Control Panel
------------
Added a really basic acp menu that doesn't really do much at the moment. To give yourself admin access, simply set your 'uploaderstatus' to 99 in the database and navigate to the preferences page from the top right corner once you login. This will allow you to delete torrents from any user.

If your 'uploaderstatus' is not set to 99 you will receive a regular 'Account Control Panel' page

AzukaChan committed:
Feature Updates

Torrent Data
----------------
Added a button to manually refresh torrent seed/leech data. There is currently no limit on this so you probably shouldn't go spamming it.


Category Overview
-----------------
Added a 'Category Overview' which can be accessed by clicking a category from the index page. Once a category reaches 10 torrents; previous/next buttons will appear at the bottom of the page allowing users to navigate through pages.


Basic Account/Admin Control Panel
------------
Added a really basic acp menu that doesn't really do much at the moment. To give yourself admin access, simply set your 'uploaderstatus' to 99 in the database and navigate to the preferences page from the top right corner once you login. This will allow you to delete torrents from any user.

If your 'uploaderstatus' is not set to 99 you will receive a regular 'Account Control Panel' page

AzukaChan committed:
Feature Updates

Torrent Data
----------------
Added a button to manually refresh torrent seed/leech data. There is currently no limit on this so you probably shouldn't go spamming it.


Category Overview
-----------------
Added a 'Category Overview' which can be accessed by clicking a category from the index page. Once a category reaches 10 torrents; previous/next buttons will appear at the bottom of the page allowing users to navigate through pages.


Basic Account/Admin Control Panel
------------
Added a really basic acp menu that doesn't really do much at the moment. To give yourself admin access, simply set your 'uploaderstatus' to 99 in the database and navigate to the preferences page from the top right corner once you login. This will allow you to delete torrents from any user.

If your 'uploaderstatus' is not set to 99 you will receive a regular 'Account Control Panel' page

KevinJDurant commented on en/view/acp_delete_file.php:
```diff
@@ -0,0 +1,11 @@
+
+	$db = new Db();
+
+	$db->query("DELETE FROM `torrents` WHERE `id`=".$_GET['delete_id']."");
```
What's the security on this? Can anyone delete any torrent file?

KevinJDurant replied:
I believe I had a simple function 'valid_user' in upload.php to see if the logged in user has a valid session key. It's at least some security, I didn't know much about this when I originally made this project.

AzukaChan replied:
Only users with 'uploaderstatus'=99 will be able to delete torrents from the acp menu. This is still very basic; I haven't added previous next buttons so the page might be huge depending on how many torrents you have on your site. Will add it on the next commit.

For anyone other than an admin; torrents will not appear on acp as they can delete their own torrents from my-torrents.php

AzukaChan replied:
Ahh yes I see; adding that would make things a little bit more secure.
Would you be able to suggest edit those things for me? I'm learning PHP as I go with this project.

KevinJDurant replied:
@AzukaChan I'll check it out during the week, busy schedule. I'll see which kind of security we can set on this API. If you want and you use Discord, you can always mail me your tag#numbers to contact@kevindurant.be.

KevinJDurant commented on en/view/category.php:
```diff
@@ -0,0 +1,278 @@
+        <footer>
+            <div class="row">
+                <div class="col-lg-12">
+                    <p>Happy Downloading~</p>
```
Copyright &copy; Your Website 2019

AzukaChan replied:
Changed 'Happy Downloading~' to 'Copyright © Your Website 2019'

KevinJDurant commented:
I see you have solved some branding issues in the next commit. But I believe you've missed some, sorry for being this strict. I can solve it myself but I only have time early next week.

Thanks for the amazing contributions so far!

AzukaChan committed:
Update category.php

Branding changes for kevin <3

AzukaChan committed:
Update user-torrents.php

Changed 'happy downloading~' to 'Copyright &copy; Your Website 2019'

AzukaChan committed:
Delete acp_delete_file.php

no longer required.

AzukaChan committed:
Account Control Panel Update

Administrator control panel is somewhat useful now;
1)Displays useful user info in a table
2) Options for changing users uploaderstatus
3) Option Removing All User Torrents (incase of flood)
4) Option for banning user (sets uploaderstatus to -1 for now?) doesn't actually do anything else for the moment.

AzukaChan commented:
http://i.imgur.com/jXDEeEv.png

Users with uploaderstatus 99 won't appear in the list just to avoid accidentally removing your own admin status.

AzukaChan committed:
Update preferences.php

Quick bug fix. Tested on accounts with different 'uploaderstatus' and everything looks good but you should still probably check it out though.
